benchmarks/Main.hs

```diff
@@ -87,12 +87,13 @@ recallPrefixed65535 =
 
 formatExponential :: Int -> Double -> String
 formatExponential precision d =
-    show (roundToPrec sig10) ++ "e" ++ show exp10Int
+  show (roundToPrec sig10) ++ "e" ++ show (exp10Int :: Int)
   where
-    roundToPrec d = let p = 10 ^^ precision
-                        n :: Integer
-                        n = round $ d * p
-                    in (realToFrac n) / p
+    roundToPrec d' =
+      let p = 10 ^^ precision
+          n :: Int
```

Comment:
Was leaving this an `Integer` causing a compiler warning or error? I'm curious what it was causing...especially because `Int` isn't arbitrary precision.

Reply:
I'm not *positive* we need it to be an integer here but it's definitely safer for `n` to be an `Integer` given that we're multiplying it by `p` which is raised to some other power...it also is probably used in the type inference for `p` since that one isn't specified.